ansible_ai_connect/ai/api/model_pipelines/langchain/pipelines.py

```diff
@@ -63,22 +63,57 @@
 HUMAN_MESSAGE_TEMPLATE = "{prompt}"
 
 
-def unwrap_playbook_answer(message: str | BaseMessage) -> tuple[str, str]:
-    task: str = ""
+def message_to_string(message: str | BaseMessage) -> str:
+    if isinstance(message, str):
+        # Ollama currently answers with just a string
+        return message
+
     if isinstance(message, BaseMessage):
         if (
             isinstance(message.content, list)
             and len(message.content)
             and isinstance(message.content[0], str)
         ):
-            task = message.content[0]
+            return message.content[0]
         elif isinstance(message.content, str):
-            task = message.content
-    elif isinstance(message, str):
-        # Ollama currently answers with just a string
-        task = message
-    if not task:
-        raise ValueError
+            return message.content
+
+    raise ValueError
+
+
+def unwrap_role_answer(message: str | BaseMessage, create_outline: bool) -> tuple[str, list, str]:
```

Comment:
I get this answer from Ollama with Granite (`granite-code:3b`):

```
Role name: nginx_setup

- name: Install Nginx
  apt:
    name: nginx

- name: Start Nginx Service
  service:
    name: nginx
    state: started

- name: Enable Nginx on Boot
  service:
    name: nginx
    enabled: yes
```

The parsing is not working.

Reply:
Another example:

```
Role name: nginx_setup
---
# main.yml

- name: Install Nginx
  apt:
    name: nginx
    state: present

- name: Enable and start Nginx service
  systemd:
    name: nginx
    enabled: yes
    state: started

defaults/main.yml
---
# defaults/main.yml

nginx_package: nginx
service_name: nginx
```

Reply:
Looks like there should be two different requests, so parsing will be easier. I will try that.

Reply:
Also, first example looks totally wrong, it didn't provide headers and the `defaults/main.yml` at all

Reply:
Granite-3b is not failing parsing anymore, but result it provides is mediocre, still suggesting to use `mistral:instruct` instead.

But parsing works now, thank you for the catch Goneri 👍